### ISSUE TITLE
[BUDI-7243] Pagination not shown in published application but works fine in build mode

### ISSUE THREAD

**Issue by yitecdeveloper** [Budibase/budibase#11102]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

- Cloud
  - Tenant ID: yitecai

**Describe the bug**
Pagination not showing or completely hidden in published app, but it works perfectly fine in build mode, at "Design" tab.

**To Reproduce**
1. Go to Design tab, add a table or repeater view and enable pagination
2. Try to navigate through pages to check if it's working
3. Publish the app
4. Go to the app published link

**Expected behavior**
Pagination should shown as in design mode
  
**App Export**
If possible - please attach an export of your budibase application for debugging/reproduction purposes.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome
 - Version: Chromium Engine Version 114.0.5735.198

<sub>[BUDI-7243](https://linear.app/budibase/issue/BUDI-7243/pagination-not-shown-in-published-application-but-works-fine-in-build)</sub>

**Comment by melohagan** [comment]
Hey @yitecdeveloper 

Which database are you using?

**Comment by yitecdeveloper** [comment]
I'm using REST API. My RestAPI call to my MongoDB. The reason I can't use Mongo directly is because if I do, budibase can't even enable pagination in build mode

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by melohagan** [comment]
Thanks @yitecdeveloper How do you have your pagination configured in the REST connector? Could you provide some screenshots please.

**Comment by yitecdeveloper** [comment]
![Screenshot 2023-07-03 at 19 13 03](https://github.com/Budibase/budibase/assets/129858833/4d052bf6-3000-4311-85e9-454d3e3b8500)
Sure did, It works perfectly in build mode. I can navigate normally in build mode, so it should working without problem. But after publish, it's gone!

**Comment by melohagan** [comment]
Hey @yitecdeveloper 

I have tried to replicate this one, but in the published app the pagination appears to be working as expected. Tried both a Table block and Repeater block.

Going to close this one off for now.

**Comment by valentinschabschneider** [comment]
this is also happening in our app

```
{
  "browser": {
    "language": "en-GB",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0",
    "platform": "Win32",
    "vendor": ""
  },
  "server": {
    "budibaseVersion": "2.11.16",
    "hosting": "docker-compose",
    "nodeVersion": "v18.18.0",
    "platform": "linux",
    "cpuArch": "x64",
    "cpuCores": 8,
    "cpuInfo": "Intel Xeon Processor (Skylake, IBRS)",
    "totalMemory": "30.67550277709961GB",
    "uptime": "356 day(s), 4 hour(s), 49 minute(s)"
  }
}
```

the data that is displayed in the published table is the one of the first page. so it seems to paginate but the page selection arrows are not shown. i can confirm that they are not in the html dom.

**Comment by aptkingston** [comment]
There is no difference between the preview or published app concerning whether pagination is shown.

There are only 2 conditions:
- You have enabled the pagination setting
- You are using a datasource that supports pagination

There's no logic that would behave any differently in the preview compared to published apps.

I've just tested this myself to double check, using a REST API query with pagination, and it works as expected in both the preview and the published app. Tested with both a table block and a data provider.

We will need an app export that has this issue in order to investigate this, or at least exact replication steps as we aren't able to reproduce this.

**Comment by RyanWang1982** [comment]
I got the same issue, even after I update the container to version 2.13.3.
I can be sure that the datasource supports pagination, cause the pagination is working fine in Postman.
I find that the data provider "paginate" did not send any parameters in published, but working fine in build or preview model.
In preview: 
<img width="294" alt="Preview" src="https://github.com/Budibase/budibase/assets/24889551/1d027fb8-534d-4607-8c08-b10ab501f8c6">
In published:
<img width="286" alt="Published" src="https://github.com/Budibase/budibase/assets/24889551/779d16e0-ff64-43f4-b8ec-f8969cc8d74a">

**Comment by RyanWang1982** [comment]
> I got the same issue, even after I update the container to version 2.13.3. I can be sure that the datasource supports pagination, cause the pagination is working fine in Postman. I find that the data provider "paginate" did not send any parameters in published, but working fine in build or preview model. In preview: <img alt="Preview" width="294" src="https://user-images.githubusercontent.com/24889551/281282733-1d027fb8-534d-4607-8c08-b10ab501f8c6.png"> In published: <img alt="Published" width="286" src="https://user-images.githubusercontent.com/24889551/281284042-779d16e0-ff64-43f4-b8ec-f8969cc8d74a.png">

And in published app, it request the same datasource twice while in preview app only once.

**Comment by RyanWang1982** [comment]
<img width="959" alt="2023-11-08_145740" src="https://github.com/Budibase/budibase/assets/24889551/63cf9bea-1f42-4e65-90d9-b8ccd65f894b">
Check this.

**Comment by aptkingston** [comment]
Thanks @RyanWang1982. That screenshot has been really helpful.

**Comment by aptkingston** [comment]
Just to clarify that this is indeed a valid bug then - query definitions are different in published apps and one of the values  removed is the `fields` property which contains the pagination info. I'll investigate why these fields are being stripped.

**Comment by Freundschaft** [comment]
im encountering the same issue, in preview mode pagination is there, in published app its missing for REST API

**Comment by m-halawany** [comment]
@Freundschaft Good catch, I've tried to duplicate the API datasource and using the new duplication on the data provider and it works for me
